ssl/ssl_conf.c

```diff
@@ -597,10 +597,10 @@ static int cmd_DHParameters(SSL_CONF_CTX *cctx, const char *value)
             = OSSL_DECODER_CTX_new_for_pkey(&dhpkey, "PEM", NULL, "DH",
                                             OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS,
                                             sslctx->libctx, sslctx->propq);
-        if (decoderctx == NULL
-                || !OSSL_DECODER_from_bio(decoderctx, in)) {
-            OSSL_DECODER_CTX_free(decoderctx);
+        if (decoderctx == NULL)
             goto end;
+        while (!OSSL_DECODER_from_bio(decoderctx, in) && !BIO_eof(in)) {
+            ERR_clear_error();
```

Comment:
Please do not use `ERR_clear_error();` that should be avoided in the library if possible. Either you could enclose the whole loop in `ERR_set_mark();` `ERR_pop_to_mark();` pair, but that would clear errors even if no params are found at all. I would do it so you set the mark at the beginning and then exit the loop if  `dhpkey != NULL` - in that case I'd call `ERR_pop_to_mark();` otherwise if the exit is due to OSSL_DECODER_from_bio() failure or BIO_eof() I would call `ERR_clear_last_mark();`